sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/service/LightWalletGrpcService.kt

```diff
@@ -67,11 +67,13 @@ class LightWalletGrpcService private constructor(
 
     /* LightWalletService implementation */
 
-    override fun getBlockRange(heightRange: ClosedRange<BlockHeight>): List<CompactFormats.CompactBlock> {
-        if (heightRange.isEmpty()) return listOf()
+    override fun getBlockRange(heightRange: ClosedRange<BlockHeight>): Sequence<CompactFormats.CompactBlock> {
+        if (heightRange.isEmpty()) {
+            return emptySequence()
+        }
 
         return requireChannel().createStub(streamingRequestTimeout)
-            .getBlockRange(heightRange.toBlockRange()).toList()
+            .getBlockRange(heightRange.toBlockRange()).iterator().asSequence()
```

Comment:
A Kotlin sequence is a stream, so this should be an improvement over direct conversion to List

Reply:
is this waiting on the stream to complete to create the sequence, or is this the actual stream conforming to the Sequence interface?

Reply:
This does assume that gRPC is doing the right thing when it exposes an Iterator.

Kotlin Sequences are generally the right lazy interface to use.